### DISCUSSION TITLE
ProjectMetricsUpdateTask causes high CPU usage

### DISCUSSION THREAD

**Discussion by navanin** [DependencyTrack/dependency-track#3905] [Q&A]
Good afternoon!
We are facing a high CPU load problem due to the **ProjectMetricsUpdateTask**. Apparently, this task is run every hour for each project, which causes a huge load on the processor, which leads to unavailability of the service and alerts from monitoring. 

As I understand it, this is a kind of update of vulnerability data in existing projects. Can we optimize this task or disable it permanently?

_DependencyTrack version: 4.11.0 (running by docker-compose)_
_OS version: Ubuntu 22.04_

**Comment by navanin** [comment]
If I can provide any more useful information, please let me know and I will do it right away

**Comment by nscuro** [comment]
You can extend the cadence at which the task executes via *Administration* -> *Configuration* -> *Task Scheduler* -> *Portfolio Metrics Update*. Note that changing this setting will require a restart.

It's odd that it causes such excessive load though. Can you share more details about how many resources you assigned to the VM / Pod that runs DT?

**Reply by navanin** [reply]
Thank you for the answer! 

I saw a mention of Portfolio Metrics Update in the project documentation, however, cadence is set to 3h, but the task still starts an hour after the container is launched. 

Of course, I can share the VM configuration:
CPU - 6 vCPU 3.00Ghz
RAM - 24 GB

Is there anything else you might need to help me solve the problem? I would really appreciate!

**Reply by nscuro** [reply]
There is a hardcoded initial delay which might be one hour... After that initial delay, the cadence will take effect.

> Is there anything else you might need to help me solve the problem?

What database are you using? Is the database running on a different VM than DT? If not, is it the database or DT itself that causes the spike in CPU usage?

Are you [monitoring](https://docs.dependencytrack.org/getting-started/monitoring/#metrics) your DT application? Can you rule out a saturated database connection pool being the problem?

How many projects / components do you have in your portfolio? How long is the task running for?

**Reply by navanin** [reply]
PostgreSQL 16 is used, installed on a separate VM. The database VM shows a healthy load. 

To be honest, I did not think about the connection pool, it is likely that this will solve our problem - I will check this information and come back with an update.

**Reply by navanin** [reply]
Nope, database is fine, the problem is not with the connection pool

**Reply by navanin** [reply]
We're getting LA 7 on CPU. This is really insane. Is there really no ways to disable or limit usage of this task?

**Comment by navanin** [comment]
@nscuro, can i completely disable this task?

**Reply by nscuro** [reply]
That is not supported.
